libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/BiometricAuthentication/BiometricAuthenticationManagerInternal.swift

```diff
@@ -81,12 +81,16 @@ public class BiometricAuthenticationManagerInternal: NSObject, BiometricAuthenti
     }
     
     @objc public func storePolicy(userAccount: UserAccount, hasMobilePolicy: Bool, sessionTimeout: Int32) {
-        let policyData = try! JSONEncoder().encode(
-            BioAuthPolicy(hasPolicy: hasMobilePolicy, timeout: sessionTimeout)
-        )
-        let result = KeychainHelper.write(service: kBioAuthPolicyIdentifier, data: policyData, account: userAccount.idData.userId)
-        if result.success {
-            SFSDKCoreLogger.i(BiometricAuthenticationManagerInternal.self, message: "Biometric authentication policy stored for user.")
+        let existingPolicy = readBioAuthPolicy(userId: userAccount.idData.userId)
+        if let policyData = try? JSONEncoder().encode(
+            BioAuthPolicy(hasPolicy: hasMobilePolicy, timeout: sessionTimeout, optIn: existingPolicy?.optIn)
```

Comment:
This is the bugfix.  Retain existing optIn status if it exists.  Also I don't think we should retain native button preference with user since it should be set for the app, not the user.